src/xdocs/config_filters.xml

```diff
@@ -1088,7 +1088,7 @@ public class UserService {
 &lt;/module&gt;
         </source>
         <p>
-          A <a href="/dtds/suppressions_1_2_xpath_experimental.dtd"><em>suppressions XML
+          A <a href="dtds/suppressions_1_2_xpath_experimental.dtd"><em>suppressions XML
```

Comment:
@romani Will these links exist for previous versions when the documentation moves to special directory?

Reply:
all links to DTDs are redirected to checkstyle.org, sourceforge.io do not have dtds folder - https://github.com/checkstyle/checkstyle/issues/6804#issuecomment-504636432

we have NO dtds at all in sourceforce hosting:
```
[romanivanov@shell-23001 ~]$ cd /home/project-web/checkstyle
[romanivanov@shell-23001 checkstyle]$ cd htdocs
[romanivanov@shell-23001 htdocs]$ find . -name *.dtd
[romanivanov@shell-23001 htdocs]$ 
```

Reply:
Then this change will break DTD links when they move to sourceforge's version folder.

Reply:
https://checkstyle.sourceforge.io/version/8.21/config_filters.html#SuppressionFilter search for " suppressions XML document in a file" , do click, works fine.

in code we do not have leading slash:
```
✔ ~/java/github/romani/checkstyle
$ ag -Q "href=\"dtds"
src/xdocs/config_filters.xml
440:            a <a href="dtds/suppressions_1_2.dtd">suppressions XML
483:          A <a href="dtds/suppressions_1_2.dtd">suppressions XML

src/xdocs/config_imports.xml
964:          href="dtds/import_control_1_4.dtd">

src/xdocs/config.xml
935:        loads the document. DTD is available at <a href="dtds/packages_1_0.dtd">packages_1_0.dtd</a>
989:        <a href="dtds/configuration_1_3.dtd">latest DTD</a>,

```

Reply:
Issue will only become apparent when this PR is merged and then that site becomes an old version.

Existing links only work if they have beginning slash. The ones without a beginning slash do not work. See https://checkstyle.sourceforge.io/version/8.26/config_imports.html#ImportControl_Description as an example from your search. Clicking on it sends user to 404.

The reason why the beginning slash makes a difference is removing it changes the URL from a absolute path to a relative path. In base directory of `https://checkstyle.sourceforge.io` absolute and relative are the same. However when switching the directory we are accessing from via `https://checkstyle.sourceforge.io/version/8.26/` for versions, absolute path forces us back to `https://checkstyle.sourceforge.io` while relative continue using `https://checkstyle.sourceforge.io/version/8.26/`.

DTDs are located at https://checkstyle.sourceforge.io/dtds/import_control_1_4.dtd and not https://checkstyle.sourceforge.io/version/8.26/dtds/import_control_1_4.dtd .

Reply:
so it does mean that ideally we should update all others to be absolute path URLs. Right ?

we can do this in separate issue/PR, and probably some regexp Check should be created to keep this nuance active in validation.

Reply:
Right. This highlighted item also needs to be reverted.

Reply:
done. Now all DTD links starts with "/dtds/"